### DISCUSSION TITLE
Thread device's Channel, PAN ID, XPAN ID, and Network Name

### DISCUSSION THREAD

**Discussion by OlivierGre** [openthread/openthread#7857] [Q&A] [answered]
Hi,
In Thread documentation ([link](https://openthread.io/guides/thread-primer/network-discovery#join_an_existing_network)) there is:

> If the device elects to join an existing network, it configures its Channel, PAN ID, XPAN ID, and Network Name to match that of the target network via Thread Commissioning, then goes through the MLE Attach process to attach as a Child (End Device). This process is used for Child-Parent links.

It confused me because, in our implementation, those parameters are set in the Thread device **before** the commissioning. If we don't, the commissioning doesn't work.
I don't see how the Thread device could change its Channel, PAN ID, XPAN ID, and Network Name to match that of the target network.

I'm also wondering how it will work for commercial products:
Let's say that you have a Thread network at home and that you buy a new Thread device. How will the user set the Channel, PAN ID, XPAN ID, and Network Name of the device to match that of the existing network?

Thank you

**Comment by jwhui** [comment, marked as answer]
> It confused me because, in our implementation, those parameters are set in the Thread device before the commissioning. If we don't, the commissioning doesn't work. I don't see how the Thread device could change its Channel, PAN ID, XPAN ID, and Network Name to match that of the target network

The Joining device should obtain **all** Thread network parameters (including Channel, Pan ID, XPAN ID, and Network Name) as part of Thread Commissioning. You should not need to configure it on the Joiner before Commissioning. See [src/cli/README_COMMISSIONING.md](https://github.com/openthread/openthread/blob/main/src/cli/README_COMMISSIONING.md)

Some products may choose to use what's called "out-of-band commissioning" for configuring Thread network parameters and not use Thread Commissioning. In this case, the only parameter the Joining device needs is the Network Key. See [src/cli/README_DATASET.md](https://github.com/openthread/openthread/blob/main/src/cli/README_DATASET.md)

**Reply by OlivierGre** [reply]
Sorry for the duplicates. I have had an error when I have pushed the first message and I haven't noticed that it had been sent.

**Reply by OlivierGre** [reply]
> The Joining device should obtain **all** Thread network parameters (including Channel, Pan ID, XPAN ID, and Network Name) as part of Thread Commissioning. You should not need to configure it on the Joiner before Commissioning. See [src/cli/README_COMMISSIONING.md](https://github.com/openthread/openthread/blob/main/src/cli/README_COMMISSIONING.md)
> 

@jwhui  Thank you very much!

I've got a setup working fine when there is a single Thread network and I would like to clarify what happens in case of multiple Thread networks.


Let's say that the Joining device is a light bulb (with a QR Code for the commissioning) and let's say that there are several Thread networks in the neighborhood (only one is "our" Thread network).

The commissioner is an Android phone and it is connected to one OTBR via Wifi.

With the Android commissioner, the user scans the QR code present on the device and press a button on the Thread device to start the network Discovery. If my understanding is correct, 2 things will then happen in parallel:

- Thanks to the QRCode data, the Commissioner gets the device unique ID ("EUI") and a passphrase ("commissioning credential"). He passes them to the OTBR to indicate that this device is going to join and should be accepted on the network.

- The Thread device discovers all the Thread Network. For each of them, it get all the network parameters that you mention (Channel, Pan ID, XPAN ID, Network Name... etc). 

From this point, I'm not 100% sure of what will happen: I expect that the device will contact each Router and ask an access. Only one of them (the one in contact with the Commissioner) will accept it. Is it correct?

**Reply by jwhui** [reply]
The Commissioner will connect to a specific Thread Network and provide that network with the list of Joiner IDs it is looking to join. In your case, the Commissioner obtains the Joiner IDs via the QR code. The Joiner ID information is encoded in the Thread Network Data in a Steering Data TLV disseminated to all Thread FTDs.

The Joiner will then scan for Thread networks that have commissioning enabled and are looking to join its Joiner ID. If the Joiner finds such a neighbor, it will attempt to establish a secure session via that neighbor to the Commissioner.

**Reply by OlivierGre** [reply]
@jwhui That's clear, thank you very much.
